README.md

```diff
@@ -125,75 +250,78 @@ Output should look like:
-'entropy_global_block': 14.501
-'entropy_lempel_ziv': 1.727
-'entropy_k_block_diff_1': -0.019
-'entropy_k_block_diff_3': 1.837
-'entropy_k_block_diff_5': 1.712
-'entropy_k_block_ratio_1': 2.262
-'entropy_k_block_ratio_3': 3.238
-'entropy_k_block_ratio_5': 2.538
-'entropy_knn_3': 4.956
-'entropy_knn_5': 4.49
-'entropy_knn_7': 4.191
-'variant_entropy': 40624.49329803771
-'normalized_variant_entropy': 0.6957588422064969
-'sequence_entropy': 76528.6794749776
-'normalized_sequence_entropy': 0.5223430410751398
-'sequence_entropy_linear_forgetting': 32139.284589305265
-'normalized_sequence_entropy_linear_forgetting': 0.21936523360299368
-'sequence_entropy_exponential_forgetting': 43880.53919110408
-'normalized_sequence_entropy_exponential_forgetting': 0.29950463593968696
+'trace_len_std': 11.470474925273926
+'trace_len_variance': 131.57179501133788
+'within_day_time_coefficient_variation': 0.49820042247168106
+'within_day_time_entropy': 9.501009299480838
+'within_day_time_geometric_mean': 35069.233548115764
+'within_day_time_geometric_std': 1.9726454507370417
+'within_day_time_harmonic_mean': 0.0
+'within_day_time_hist0': 3.2487851331575074e-06
+'within_day_time_hist1': 2.617288257157336e-06
+'within_day_time_hist2': 3.474754497220219e-05
+'within_day_time_hist3': 1.1808230741593563e-05
+'within_day_time_hist4': 1.2667979500485362e-05
+'within_day_time_hist5': 1.3687504577401301e-05
+'within_day_time_hist6': 1.1876706306461051e-05
+'within_day_time_hist7': 9.822439360436398e-06
+'within_day_time_hist8': 9.297460029785654e-06
+'within_day_time_hist9': 5.980199331760658e-06
+'within_day_time_iqr': 34486.25
+'within_day_time_kurtosis': -0.9142275965359783
+'within_day_time_kurtosis_hist': 2.6115894228132266
+'within_day_time_max': 86390.0
+'within_day_time_mean': 41330.543183909555
+'within_day_time_median': 37800.0
+'within_day_time_min': 0.0
+'within_day_time_mode': 21600.0
+'within_day_time_q1': 23113.75
+'within_day_time_q3': 57600.0
+'within_day_time_skewness': 0.3603519661740256
+'within_day_time_skewness_hist': 1.7511033515349685
+'within_day_time_std': 20590.894075207754
+'within_day_time_variance': 423984918.81642574
 }
+```
+
+## Tutorial for extending to additional features (e.g. time-based)
+
+For this tutorial, we focus on the example of time-based features. The `feeed/time.py` is a script that currently contains the class `Timestamp`, which extracts knowledge from timestamps. In summary, features could be extracted from groups (i.e., cases) or from the whole log (e.g., time within the day).
+
+Implementing each time-based feature as `@classmethods` within this class allows us to easily scale and manage features. A tiny botleneck is that each class method should accept `**kwargs` regardless of the other arguments (but this can be internaly handled in the future). Each class method is accessed by inspecting the object using `inspect.getmembers`. 
+
+All the features are currently measured in seconds, and they include include:
```

Comment:
include include

Reply:
I think mentioning these for classes here ist too specific. If we do mention them, please move that to feeed/time.py. This tutorial should remain general.